components/prompt_based_laion_retrieval/src/main.py

```diff
@@ -70,33 +82,20 @@ def transform(
             modality=Modality.IMAGE,
         )
 
-        print("Input dataframe:", dataframe.head(2))
+        meta_df = pd.DataFrame(columns=["id", "url"], dtype="string")
+        meta_df = meta_df.set_index("id")
 
-        logger.info("Retrieving URLs...")
-        dataframe["images_url"] = dataframe["prompts_text"].apply(
-            lambda example: query_clip_client(example, client),
-            meta=("images_url", "str"),
+        dataframe = dataframe.map_partitions(
```

Comment:
Maybe it's worth it calculating the cost of deploying LAION ourself in an endpoint somewhere. If the cost is reasonable it would avoid the timeouts of the LAION-hosted dataset

Reply:
Yes, or if we could help host the LAION one. Something to look into.

BTW, this PR is outdated.
We already split these parts into separate PRs:

- Refactor the DataWriter class to only call compute() once.
- Remove the id and source column and let the user work with the dataframe index directly. The separate columns were giving too many issues without benefits. It was not clear when the uid column should be rebuild and reset, which led to every component both resetting and rebuilding it.

Still need to do the rest.